### ISSUE TITLE
Window Position

### ISSUE THREAD

**Issue by smaragdus** [erickutcher/httpdownloader#4]
In fact **HTTP Downloader** remembers window size and columns' width but it does not remember window position on desktop (in my case- centered) so I have to reset it any time I start the program. It would be nice if **HTTP Downloader** can also save window position in the configuration file ('http_downloader_settings').

**Comment by erickutcher** [comment]
It should already be saving the position. I tested it on Windows 7 and Windows 10 and it works fine. Try deleting your settings file.

Before the program reads any settings and displays the main window, it'll set the x and y position variables to whatever values would make the main window centered on the screen. If it finds the saved x and y positions in the settings file, then it'll use those instead.

Settings are only saved when the program is closed/exited cleanly. Killing the process, for example, will prevent any changed settings from being saved.

**Comment by smaragdus** [comment]
@erickutcher 

I did several tests with different positions on desktop. You are right- when I set the position manually it is saved correctly. But if I use a special tool to set the position (in my case- [SmartSystemMenu](https://github.com/AlexanderPro/SmartSystemMenu)), the position is not saved, screen:

![http downloader 1 0 0 6 - 2018-09-26 - 001](https://user-images.githubusercontent.com/9025757/46068875-d946b880-c182-11e8-8514-7cd2824293aa.png)

In fact I was using this tool (**SmartSystemMenu**) trying to set the window position and that was why I opened this issue. I tried another tool ([AutoSizer](http://www.southbaypc.com/AutoSizer/))- the same result.

In short:

- if I set the window position manually **HTTP Downloader** saves it;
- if I set the window position with the help of external tool **HTTP Downloader** does not save it;

For me this is strange since I have been using **SmartSystemMenu** for settings windows positions for years and for some mysterious reason **HTTP Downloader** ignores it. If I am not asking too much you may check **HTTP Downloader** with **SmartSystemMenu** to confirm/negate the issue.

**Comment by erickutcher** [comment]
The SmartSystemMenu is sending a MoveWindow call to my program. Unfortunately the messages that are triggered by it don't have any code that saves the changes.

I implemented the necessary detection and it should save things properly now.

**Comment by smaragdus** [comment]
@erickutcher 

Thank you for taking your time to investigate the problem fix it!

**Comment by smaragdus** [comment]
@erickutcher 

I can confirm that window positioning with external tools now works just fine, thanks!
